iceoryx_dds/include/iceoryx_dds/internal/gateway/iox_to_dds.inl

```diff
@@ -110,8 +111,15 @@ inline void Iceoryx2DDSGateway<channel_t, gateway_t>::forward(const channel_t& c
     {
         subscriber->take().and_then([&](const void* userPayload) {
             auto dataWriter = channel.getExternalTerminal();
-            auto header = iox::mepoo::ChunkHeader::fromUserPayload(userPayload);
-            dataWriter->write(static_cast<const uint8_t*>(userPayload), header->userPayloadSize());
+            auto chunkHeader = iox::mepoo::ChunkHeader::fromUserPayload(userPayload);
+            iox::dds::IoxChunkDatagramHeader datagramHeader;
```

Comment:
the datagram header is used only in the gateway and doesn't fit into the ChunkHeader. I could add a conversion ctor/method to IoxChunkDatagramHeader though. Since time is running I would consider this a nice to have if you don't mind

Reply:
added to #376